File: test_data/code/java/1417-reformat-the-string.java
class Solution {
    public String reformat(String s) {
        int letterCount = 0;
        int digitCount = 0;
        for (char c : s.toCharArray()) {
            if (Character.isLetter(c)) {
                letterCount++;
            } else {
                digitCount++;
            }
        }
        if (Math.abs(letterCount - digitCount) > 1) {
            return "";
        }
        StringBuilder result = new StringBuilder();
        int letterPtr = 0;
        int digitPtr = 0;
        for (char c : s.toCharArray()) {
            if (Character.isLetter(c)) {
                if (digitPtr < digitCount) {
                    result.append(c);
                    digitPtr++;
                } else {
                    result.append(c);
                    letterPtr++;
                }
            } else {
                if (letterPtr < letterCount) {
                    result.append(c);
                    letterPtr++;
                } else {
                    result.append(c);
                    digitPtr++;
                }
            }
        }
        return result.toString();
    }
}

File: test_data/code/python/1417-reformat-the-string.py
class Solution:
    def reformat(self, s: str) -> str:
        char_count = [0, 0]
        for char in s:
            if char.isdigit():
                char_count[0] += 1
            else:
                char_count[1] += 1
        if abs(char_count[0] - char_count[1]) > 1:
            return ""
        result = []
        if char_count[0] > char_count[1]:
            prev = 'd'
        else:
            prev = 'l'
        for char in s:
            if char_count[0] > char_count[1]:
                if char.isdigit():
                    result.append(char)
                    char_count[0] -= 1
            else:
                if char.isalpha():
                    result.append(char)
                    char_count[1] -= 1
            if prev == 'd' and char.isdigit():
                prev = 'l'
            elif prev == 'l' and char.isalpha():
                prev = 'd'
        return ''.join(result)